src/ui/utils/preferredLocation.ts

```diff
@@ -1,18 +1,33 @@
 import { MappedLocation } from "@replayio/protocol";
 import { ThreadFront } from "protocol/thread";
 import { UIStore } from "ui/actions";
-import { getPreferredLocation as getPreferredLocationSelector } from "ui/reducers/sources";
+import {
+  getPreferredLocation as getPreferredLocationSelector,
+  getCorrespondingSourceIds,
+} from "ui/reducers/sources";
 
 // TODO [hbenl] some components ask for preferred locations in render functions or effects
 // (so without direct access to the redux store), this is a temporary hack until these
 // components are rewritten to use the sources selectors directly.
 
 let store: UIStore;
 
-export function getPreferredLocation(location: MappedLocation | undefined) {
-  return (
-    location &&
-    getPreferredLocationSelector(store.getState(), location, ThreadFront.preferredGeneratedSources)
+export function getPreferredLocation(locations: MappedLocation | undefined) {
+  if (!locations) {
+    return;
+  }
+  const state = store.getState();
+  // TODO [hbenl] another hack: the new console doesn't update sourceIds in locations
+  // to their first corresponding sourceId, which getPreferredLocationSelector
+  // subsequently complains about
+  const correspondingLocations = locations.map(location => ({
+    ...location,
+    sourceId: getCorrespondingSourceIds(state, location.sourceId)[0],
+  }));
```

Comment:
What's needed to make this _actually_ happen?

There was a Linear issue that got closed by a temporary fix but do we have a non-temporary issue open that describes what needs to be done?

Reply:
I wanted to discuss in one of our frontend meetings how we handle corresponding sources in the new architecture. Personally I think we should do what the old architecture did: update all locations as soon as we receive them from the backend, but I'm open to other solutions.